src/lib/helpers.ts

```diff
@@ -8,3 +21,47 @@ export function getUsdcAddress(networkId: number): string {
 
   return buyTokenAddress;
 }
+
+export async function sendGaslessTransaction(contractCall: PromiEvent<Contract>): Promise<any> {
+  let hashOutcome = OUTCOMES.INITIAL;
+
+  const transactionHash = await new Promise(
+    (resolve, reject) => {
+      contractCall.on('error', (error: Error) => {
+        if (hashOutcome === OUTCOMES.INITIAL) {
+          hashOutcome = OUTCOMES.REJECTED;
```

Comment:
So if `hashOutcome` wasn't previously assigned in the `contractCall.on('transactionHash'` callback function, then it will enter this if-statement?

Reply:
ya but I think its primarily for if we've hit the error call already